registry/storage/driver/azure/parser.go

```diff
@@ -0,0 +1,54 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	defaultRealm = "core.windows.net"
+)
+
+type Credentials struct {
+	Type     string `yaml:"type"`
+	ClientID string `yaml:"clientId"`
+	TenantID string `yaml:"tenantId"`
+	Secret   string `yaml:"secret"`
+}
+
+type Parameters struct {
+	Container        string      `yaml:"container"`
+	AccountName      string      `yaml:"accountname"`
+	AccountKey       string      `yaml:"accountkey"`
+	Credentials      Credentials `yaml:"credentials"`
+	ConnectionString string      `yaml:"connectionstring"`
+	Realm            string      `yaml:"realm"`
+	RootDirectory    string      `yaml:"rootdirectory"`
+	ServiceURL       string      `yaml:"serviceurl"`
+}
+
+func NewParameters(parameters map[string]interface{}) (*Parameters, error) {
+	data, err := yaml.Marshal(parameters)
+	if err != nil {
+		return nil, err
+	}
+	var res Parameters
+	if err := yaml.Unmarshal(data, &res); err != nil {
+		return nil, err
+	}
```

Comment:
I'm not a huge fan of using libraries outside the core golang modules unless you have to.  And it felt like since you already use yaml for the config parsing, it seemed like a trivial one-time roundtrip makes sense.  But happy to reconsider if you like.